library/std/src/sys/pal/unix/process/process_unix.rs

```diff
@@ -871,6 +865,42 @@ impl Command {
     }
 }
 
+type PosixSpawnAddChdirFn =
+    unsafe extern "C" fn(*mut libc::posix_spawn_file_actions_t, *const libc::c_char) -> libc::c_int;
+
+/// Get the function pointer for adding a chdir action to a
+/// `posix_spawn_file_actions_t`, if available, assuming a dynamic libc.
+///
+/// Some platforms can set a new working directory for a spawned process in the
+/// `posix_spawn` path. This function looks up the function pointer for adding
+/// such an action to a `posix_spawn_file_actions_t` struct.
+#[cfg(not(all(target_os = "linux", target_env = "musl")))]
+fn get_posix_spawn_addchdir() -> Option<PosixSpawnAddChdirFn> {
```

Comment:
Unfortunately, you've moved this to the top of scope in a sensible-seeming fashion. Unfortunately, this is only called within `posix_spawn`, which not every platform calls to begin with. Indeed, many `cfg` it away! So this will not be called at all on platforms that don't match this cfg:
```rust
    #[cfg(any(
        target_os = "freebsd",
        all(target_os = "linux", target_env = "gnu"),
        all(target_os = "linux", target_env = "musl"),
        target_os = "nto",
        target_vendor = "apple",
    ))]
```

But it will still be defined for those. For those which aren't in the above set, but do use `process_unix.rs`, then they will see an unused function, and then libstd will not build.

Reply:
Ah, right...

I guess I could probably move the function item to be function-local.

Reply:
That's probably easiest, yeah.

Reply:
All right, updated, that seems to work.